lib/src/main/java/ECSE428Project/service/AccountService.java

```diff
@@ -35,6 +36,29 @@ public Account createAccount(String name, String email, String password) {
         account = accountRepository.save(account);
         return account;
     }
+    
+  @Transactional
+  public Account changePassword(String id, String oldPass, String newPass) {
+    Optional<Account> opt = accountRepository.findById(id);
+    Account account;
+    
+    if (opt.isPresent()) {
+      account = opt.get();
+      
+      //if the oldpassword is correct, we make the change, otherwise the password stays the same
+      if (account.getPassword().equals(oldPass)) {
+        account.setPassword(newPass);
+        account = accountRepository.save(account);
+      }
+
```

Comment:
Maybe add and elseif here: throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "The password is incorrect");
rather than in the controller method
I dont know which one is best practice though I feel like you would know better...

Reply:
Yeah it doesnt rly matter let's keep it as is